pkg/workload/workload.go

```diff
@@ -309,6 +309,7 @@ func UnsetQuotaReservationWithCondition(wl *kueue.Workload, reason, message stri
 	}
 	apimeta.SetStatusCondition(&wl.Status.Conditions, condition)
 	wl.Status.Admission = nil
+	_ = SyncAdmittedCondition(wl)
```

Comment:
Can you explain why this is needed? Maybe a comment would be nice.

Reply:
it will set the `Admitted` cond to false if it is true. Comment added.